plugins/alignments/src/LinearPileupDisplay/components/ColorByCustom.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react'
+import { observer } from 'mobx-react'
+import { Button, DialogContent, DialogActions, Typography } from '@mui/material'
+import { Dialog } from '@jbrowse/core/ui'
+import { ColorPicker } from '@jbrowse/core/ui/ColorPicker'
+
+import { fillColor, IColorByModel } from '../../shared/color'
+
+function ColorByCustomDlg(props: {
+  model: {
+    colorBy?: IColorByModel
+    setColorScheme: Function
+  }
+  handleClose: () => void
+}) {
+  const { model, handleClose } = props
+
+  const type = model.colorBy ? model.colorBy.type : 'normal'
+  const tag = model.colorBy ? model.colorBy.tag : ''
+  const extra = model.colorBy ? model.colorBy.extra : undefined
+
+  const colorOptionMap = {
```

Comment:
@cmdcolin 
> is it meant that "Custom color scheme" changes just "color_unknown" and then color_unknown is immediately applied to all reads?

It should detect the "colorScheme" the track is set to e.g. normal, strand, etc. and allow the user to select a colour for each corresponding field to that scheme. If it's set to "normal" the only scheme available is color_unknown...can change if that's incorrect but that's what I perceived from the codebase.

The colours available to select are mapped here:

Reply:
When I perform a "Restore Default", the color is darker than it was before changing it. I think that is because it goes from "normal" coloring (light grey) to unknown (darker grey) in that process. there may want to be a color_normal or something

Reply:
restore default maybe also can "clear" any color setting

Reply:
^ that's what its supposed to do, i think with 'normal' i just have it mapped to the wrong setting, i.e. unknown instead of whatever its supposed to be

Reply:
I changed unknown to be lightgrey, which was what was happening -- since normal was the default it was hitting another switch case that sent it to lightgrey. since unknown isn't used otherwise I think this is ok